storm-client/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -250,11 +254,14 @@ private String metricName(String name, String stormId, String componentId, Integ
         sb.append(workerPort);
         sb.append("-");
         sb.append(name);
-        return sb.toString();
+        String longName = sb.toString();
+        String shortName = WORKER_METRIC_PREFIX + name;
```

Comment:
I am trying to understand why we have `storm.worker` and `storm.topology` prefix. Do we need there prefix on shortnames?

Reply:
I thought @kishorvpatil was in favor of keeping these.  I am fine dropping them for short or long names if desired.  Let me know what you want.

Reply:
i think we can leave it as _storm.topology,_  similar to nimbus metrics and supervisor metrics. Also, storm.worker is suggesting they are worker level metrics.

Reply:
@Ethanlm - just making sure you're ok with the prefixes....

Reply:
I took a closer look at what `storm.worker` and `storm.topology` really are. So in the original code, when the metric is registered using TopologyContext, it will prefix with `storm.topology`, everything else is `storm.worker`. And there are two formats for `storm.worker`, one with streamId, the other without streamId.

However, `storm.topology` and `storm.worker` doesn't really mean anything.

The metric registered at TopologyContext is still task level metric, which corresponds to specific taskIds. And metrics prefixed with `storm.worker`, currently including 

1. JCQueue Metrics: executor-transfter-queue metric (executor level), worker-transfer-queue metric (worker level)
2. TaskMetrics: __acker, __failed, etc, are all taskId + streamId level. 

So these `storm.worker` and `storm.topology` don't really mean anything and keeping it will just cause confusion. 

It is okay to not deal with it in this PR. But we shouldn't add prefix to old V1 metrics, which unnecessarily breaks backwards-compatibility. 

Another issue we probably need to change the long name format to be something like 
`topoId.<topoId>.hostname.<hostname>...`
so a reporter can parse the metric into dimensions. Our current format of long metrics is not easy to parse. This can be a follow-up work.

Reply:
I will remove the prefixes for the short names for the purpose of this JIRA.